xilem_core/src/views/fork.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use core::marker::PhantomData;
+
+use crate::{
+    AppendVec, ElementSplice, Mut, NoElement, View, ViewId, ViewPathTracker, ViewSequence,
+};
+
+/// Create a view which acts as `active_view`, whilst also running `alongside_view`, without inserting it into the tree.
+///
+/// `alongside_view` must be a `ViewSequence` with an element type of [`NoElement`].
+pub fn fork<Active, Alongside, Marker>(
+    active_view: Active,
+    alongside_view: Alongside,
+) -> Fork<Active, Alongside, Marker> {
+    Fork {
+        active_view,
+        alongside_view,
+        marker: PhantomData,
+    }
+}
+
+/// The view for [`fork`].
+pub struct Fork<Active, Alongside, Marker> {
+    active_view: Active,
+    alongside_view: Alongside,
+    marker: PhantomData<Marker>,
+}
+
+impl<State, Action, Context, Active, Alongside, Marker, Message>
+    View<State, Action, Context, Message> for Fork<Active, Alongside, Marker>
+where
+    Active: View<State, Action, Context, Message>,
+    Alongside: ViewSequence<State, Action, Context, NoElement, Marker, Message>,
```

Comment:
Neat trick, to allow multiple views here. But are those `unreachable!`s really unreachable below? I guess they are because of the `ViewSequence` impl above?

Reply:
I've added an explanatory comment.